src/com/b44t/messenger/DcMsg.java

```diff
@@ -145,8 +145,9 @@ public File getFileAsFile() {
 
     // aliases and higher-level tools
     public static int[] msgSetToIds(final Set<DcMsg> dcMsgs) {
-        int   cnt = dcMsgs==null? 0 : dcMsgs.size();
-        int[] ids = new int[cnt];
+        if(dcMsgs==null)
+            return new int[0];
```

Comment:
in general, it is a good idea to always surround code blocks by curly brackets:

```suggestion
        if (dcMsgs == null) {
            return new int[0];
        }
```